pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -399,6 +400,28 @@ public synchronized PinotResourceManagerResponse updateInstance(String instanceI
     }
   }
 
+  /**
+   * Updates the tags of the specified instance ID
+   */
+  public synchronized PinotResourceManagerResponse updateInstanceTags(String instanceIdToUpdate, String tags) {
+    InstanceConfig instanceConfig = getHelixInstanceConfig(instanceIdToUpdate);
```

Comment:
run this in a compare and set loop to handle race conditions. we can wrap this function in updater

Reply:
what kind of race conditions are you thinking about? All of the methods which update this config are synchronized methods

Reply:
ok, we can handle that in another PR